### PR TITLE
Fix #235526: Update musicxml to 3.1

### PR DESCRIPTION
**Pull request by Jojo-Schmitz**
Most of this, if not all, should/could most probably go to the 2.2 branch too.

### CODE REVIEW

**Comment by lasconic** [comment]
The stable version of 3.1 should be released in the coming days. Let's wait for it.

File: mscore/exportxml.cpp [hunk 1]
```diff
@@ -5024,7 +5024,7 @@ void ExportMusicXml::write(QIODevice* dev)
       xml.setDevice(dev);
       xml.setCodec("UTF-8");
       xml << "<?xml version=\"1.0\" encoding=\"UTF-8\"?>\n";
-      xml << "<!DOCTYPE score-partwise PUBLIC \"-//Recordare//DTD MusicXML 3.0 Partwise//EN\" \"http://www.musicxml.org/dtds/partwise.dtd\">\n";
+      xml << "<!DOCTYPE score-partwise PUBLIC \"-//Recordare//DTD MusicXML 3.1 Partwise//EN\" \"http://www.musicxml.org/dtds/partwise.dtd\">\n";
```
Comment (Jojo-Schmitz) [inline mscore/exportxml.cpp]: This change may be a bit premature, as we're not yet using any of the 3.1 features.

Comment (lasconic) [inline mscore/exportxml.cpp]: I would keep it if 3.1 is entirely compatible with 3.0 and I believe it's the case.

File: mscore/schema/xml.xsd [hunk 2]
```diff
@@ -1,4 +1,4 @@
-<?xml version='1.0'?>
+﻿<?xml version='1.0'?>
```
Comment (Jojo-Schmitz) [inline mscore/schema/xml.xsd]: Not quite sure what(why this change is, but that's what I got when replacing the existing file with the one taken from https://github.com/w3c/musicxml/blob/gh-pages/schema/xml.xsd

**Comment by Jojo-Schmitz** [comment]
Agreed, let's wait for the final Version. I'll update this PR as needed if and when it happens

File: mscore/exportxml.cpp (hunk 1 above)
Comment (Jojo-Schmitz) [inline mscore/exportxml.cpp]: Keep the change or keep the file unchanged?

Comment (lasconic) [inline mscore/exportxml.cpp]: I would keep the change.

Comment (Jojo-Schmitz) [inline mscore/exportxml.cpp]: OK got it now ;-)

**Comment by lvinken** [comment]
PR looks fine to me and agree with most that has been said here. Main justification for updating the schema is preventing needless validation errors on import. Should go into 2.2 too.

File: mscore/schema/xml.xsd (hunk 2 above)
Comment (Jojo-Schmitz) [inline mscore/schema/xml.xsd]: Taking it straight from a forked repo fixed this

**Comment by Jojo-Schmitz** [comment]
There's no movement on https://github.com/w3c/musicxml since August 1st (and those changes are incorporated into this PR here), the deadline of "End of July" has passed since long.
Yet Finale is already exporting 3.1 happily since end of June...

**Comment by Jojo-Schmitz** [comment]
Just noticed that Finale has `<score-partwise version="3.1">` (i.e. add the version there) in addition to `<!DOCTYPE score-partwise PUBLIC "-//Recordare//DTD MusicXML 3.1 Partwise//EN" "http://www.musicxml.org/dtds/partwise.dtd">`, we only do the latter, should we do the former too?

**Comment by lasconic** [comment]
Probably, especially if we do use 3.1 feature in export.

**Comment by Jojo-Schmitz** [comment]
As far as I know we don't do yet, but the Finale XML I was looking at didn't seem to either, as it imported cleanly without this PR. So I've made the change now.

**Comment by Jojo-Schmitz** [comment]
Seems that once https://github.com/w3c/musicxml/pull/236 is merged, we're ready do go with this PR here

**Comment by lasconic** [comment]
Last comment on the mailing list said "this month".

**Comment by Jojo-Schmitz** [comment]
https://github.com/w3c/musicxml/pull/236 got merged

**Comment by Jojo-Schmitz** [comment]
MusicXML 3.1 is officially tagged and released now, on 7Dec17, https://github.com/w3c/musicxml/commit/8bbe8e50606bf99317a86e9e1637618d6bdd1997

**Comment by lvinken** [comment]
Also announced on http://www.musicxml.com/for-developers/, so I am in favour of merging a.s.a.p.
